### ISSUE TITLE
[java] False negative on calling run() of Runnable interface

### ISSUE THREAD

**Issue by GiantCrocodile** [pmd/pmd#728]
You will experience the same issue as in `DontCallThreadRun` rule if you call `run()` of `Runnable` interface without putting it in a `new Thread(new TestRunnable()).start();` call. See issue #565.

**Comment by jsotuyod** [comment]
This should be fixed by the changes of #596, can you please try with a 6.0.0-SNAPSHOT?

**Comment by GiantCrocodile** [comment]
I tried this @jsotuyod: `pmd -d workspace\RunnableTest\src\ -f xml -R rulesets/java/basic.xml`. I got:

```
Nov 10, 2017 6:12:06 PM net.sourceforge.pmd.cache.NoopAnalysisCache <init>
WARNUNG: This analysis could be faster, please consider using Incremental Analysis: https://pmd.github.io/pmd/pmd_userdocs_getting_started.html#incremenal-analysis
<?xml version="1.0" encoding="UTF-8"?>
<pmd xmlns="http://pmd.sourceforge.net/report/2.0.0"
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
    xsi:schemaLocation="http://pmd.sourceforge.net/report/2.0.0 http://pmd.sourceforge.net/report_2_0_0.xsd"
    version="6.0.0-SNAPSHOT" timestamp="2017-11-10T18:12:07.066">
</pmd>
```

So did I try it correctly or did I do something wrong? I never use pmd from console but within Eclipse.

**Comment by jsotuyod** [comment]
Yes, that's correct. Great then, it's fixed as expected. Thanks!

**Comment by GiantCrocodile** [comment]
I don't understand this @jsotuyod. I would expect that there is an exception/error mentioned in the output I did paste? There **should** be an error from PMD, not none.

**Comment by jsotuyod** [comment]
Sorry, I misunderstood your report.

Nontheless, I don't believe this would be correct. Calling `Runnable.run()` (or `Callable.call()` which is the non-void equivalent) is not necessarily an issue, unlike calling `Thread.run()` which is almost always a programmer mistake, since the interface is designed solely to be able to do that. Flagging these would result necessarily in lots of false positives, effectively making it impossible to use `Runnable` / `Callable` for anything other than passing it to JRE classes.

For instance, things such as writing our own `ExecutorService` implementation would be impossible without being flagged. Lots of libraries rely on this.

Just to show how broken this would be, PMD itself is calling directly `Callable.call()` on our [`MonoThreadProcessor`](https://github.com/pmd/pmd/blob/master/pmd-core/src/main/java/net/sourceforge/pmd/processor/MonoThreadProcessor.java#L29)

Therefore, not using `Runnable.run()` directly seems more like a project-specific thing that can be written into an ad-hoc XPath rule, than a general rule to be part of PMD.
